modules/cache-common/src/main/java/org/opensearch/cache/common/tier/TieredSpilloverCache.java

```diff
@@ -222,6 +229,61 @@ public V computeIfAbsent(ICacheKey<K> key, LoadAwareCacheLoader<ICacheKey<K>, V>
         return cacheValueTuple.v1();
     }
 
+    private V compute(ICacheKey<K> key, LoadAwareCacheLoader<ICacheKey<K>, V> loader) throws Exception {
+        // A future that returns a pair of key/value.
+        CompletableFuture<Tuple<ICacheKey<K>, V>> completableFuture = new CompletableFuture<>();
+        // Only one of the threads will succeed putting a future into map for the same key.
+        // Rest will fetch existing future.
+        CompletableFuture<Tuple<ICacheKey<K>, V>> future = completableFutureMap.putIfAbsent(key, completableFuture);
+        // Handler to handle results post processing. Takes a tuple<key, value> or exception as an input and returns
+        // the value. Also before returning value, puts the value in cache.
+        BiFunction<Tuple<ICacheKey<K>, V>, Throwable, V> handler = (pair, ex) -> {
+            V value = null;
+            if (pair != null) {
+                try (ReleasableLock ignore = writeLock.acquire()) {
+                    onHeapCache.put(pair.v1(), pair.v2());
+                }
+                value = pair.v2(); // Returning a value itself assuming that a next get should return the same. Should
+                // be safe to assume if we got no exception and reached here.
+            }
+            completableFutureMap.remove(key); // Remove key from map as not needed anymore.
+            return value;
+        };
+        CompletableFuture<V> completableValue;
+        if (future == null) {
```

Comment:
Will this ever be null, since the existing or computed value is never null? As per the official documentation:

Returns:
    the current (existing or computed) value associated with the specified key, or null if the computed value is null

Reply:
Sorry my bad, confused `putIfAbsent` with `computeIfAbsent`